File: backend/src/main/java/com/companyshield/demo/service/TransactionService.java
package com.companyshield.demo.service;

import com.companyshield.demo.domain.dto.TransactionDto;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.util.List;

@Service
public interface TransactionService {
    List<TransactionDto> getTransactionsForUser(String userId);
    boolean deposit(String userId, BigDecimal amount);
    boolean withdraw(String userId, BigDecimal amount);
    boolean transfer(String fromUserId, String toUserId, BigDecimal amount);
}

File: frontend/src/service/transaction_service.ts
import { GetTransactionsResponse } from "../domain/transactional/GetTransactionsResponse";
import { TransactionDto } from "../domain/TransactionDto";

export interface TransactionService {
    getTransactions: () => Promise<GetTransactionsResponse | Error>;
    deposit: (accountId: string, amount: number) => Promise<boolean>;
    withdraw: (accountId: string, amount: number) => Promise<boolean>;
    transfer: (toAccountId: string, amount: number) => Promise<boolean>;
}